clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -437,29 +456,24 @@ public void wakeup() {
      */
     @Override
     public void commitSync(final Duration timeout) {
-        final CommitApplicationEvent commitEvent = new CommitApplicationEvent(subscriptions.allConsumed());
-        eventHandler.add(commitEvent);
-
-        final CompletableFuture<Void> commitFuture = commitEvent.future();
-        try {
-            commitFuture.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
-        } catch (final TimeoutException e) {
-            throw new org.apache.kafka.common.errors.TimeoutException(
-                     "timeout");
-        } catch (final Exception e) {
-            // handle exception here
-            throw new RuntimeException(e);
-        }
+        commitSync(subscriptions.allConsumed(), timeout);
     }
 
     @Override
     public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets) {
-        throw new KafkaException("method not implemented");
+        commitSync(offsets, Duration.ofMillis(defaultApiTimeoutMs));
     }
 
     @Override
     public void commitSync(Map<TopicPartition, OffsetAndMetadata> offsets, Duration timeout) {
-        throw new KafkaException("method not implemented");
+        CompletableFuture<Void> commitFuture = commit(offsets);
+        try {
+            commitFuture.get(timeout.toMillis(), TimeUnit.MILLISECONDS);
+        } catch (final TimeoutException e) {
```

Comment:
I'm a bit unsure about the exception handling there.

Reply:
Ditto here.

Reply:
Ditto above: InterruptedException -> kafka interruptException. TimeoutException -> kafka timeoutException. ExecutionException -> kafkaException